clevertap-core/src/main/java/com/clevertap/android/sdk/network/NetworkManager.kt

```
@@ -0,0 +1,1073 @@
+            val cursor = databaseManager.getQueuedEvents(context, 50, previousCursor, eventGroup)
+
+            if (cursor.isEmpty) {
+                // No events in the queue, log and break
+                config.logger.verbose(config.accountId, "No events in the queue, failing")
+
+                if (eventGroup == EventGroup.PUSH_NOTIFICATION_VIEWED) {
+                    // Notify listener for push impression sent to the server
+                    if (previousCursor?.data != null) {
+                        try {
+                            notifyListenersForPushImpressionSentToServer(previousCursor.data!!)
+                        } catch (e: Exception) {
```

detekt (1.23.7)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)